common/weave/client.go

```diff
@@ -165,7 +169,7 @@ func (c *client) AddDNSEntry(fqdn, containerID string, ip net.IP) error {
 	req.Header.Add("Content-Length", strconv.Itoa(len(data.Encode())))
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		return err
+		return fmt.Errorf("%v. %v", err, noWeaveTip)
```

Comment:
Given how often we do this, I suggest creating an `errorf` function with the same signature as `fmt.Errorf` that appends the message.